pkg/dao/repository_configs.go

```diff
@@ -173,6 +174,21 @@ func (r repositoryConfigDaoImpl) bulkCreate(tx *gorm.DB, newRepositories []api.R
 	}
 }
 
+func (p repositoryConfigDaoImpl) InternalOnly_ListReposToSnapshot() ([]models.RepositoryConfiguration, error) {
+	var dbRepos []models.RepositoryConfiguration
+	interval := fmt.Sprintf("%v hours", config.SnapshotInterval)
+	result := p.db.Where("snapshot IS TRUE").Joins("LEFT JOIN tasks on last_snapshot_task_uuid = tasks.id").
+		Where(p.db.Where("tasks.queued_at <= (current_date - cast(? as interval))", interval).
```

Comment:
is this wrapped `where(where(...))` intentional?

Reply:
yes, it means its selecting:

Snapshot is true AND  ( "last task is old" OR "status is not complete, pending, or running" OR "Never had a snapshot task" ) 

if it wasn't wrapped, it ends up like:   Snapshot is true AND  "last task is old" OR "status is not complete, pending, or running" OR "Never had a snapshot task"   which when grouped via precedence means:

( "Snapshot is true" AND  "last task is old")  OR "status is not complete, pending, or running" OR "Never had a snapshot task"